### DISCUSSION TITLE
'man' more or less works

### DISCUSSION THREAD

**Discussion by MikeFultonDev** [zopencommunity/meta#177] [Announcements]
I wrote a [blog](https://makingdeveloperslivesbetter.wordpress.com/2023/05/29/the-wonderful-world-of-man-uals/) on how to use man: feedback welcome! 

If you want to play with 'man' so that you can see the various man pages from the tools we are building, it now works well - always more bugs to fix, but they are getting less severe.

You can install it like any other tool - `zopen install man-db` (but you need the latest meta for it to find the runtime dependencies correctly)

**NEW**: Even though the blog says it doesn't work yet, UTF-8 man pages **DO** work. 

**NEW**: If you get the latest drop, it also will gracefully 'fail over' to the system man, so you can do `man sh` and it will use the /bin/man to show the man page for 'sh'

**NEW**: if you get the latest drop (Feb 23rd) then you can run `mandb` (no options) and it will index all the man pages in your MANPATH so that you can then use `whatis` and `apropos`. You should remove the z/OS man directory from your MANPATH before running `mandb` or else you will get a warning about the z/OS man pages.

@IgorTodorovskiIBM @AnthonyGiorgio @lbdyck @gnrossi

**Comment by MikeFultonDev** [comment]
First page of 'man bash'

![image](https://user-images.githubusercontent.com/61100689/218235596-c21976ec-e6b6-481b-85ff-cdd1d984c1f1.png)

**Reply by IgorTodorovskiIBM** [reply]
Nice!

**Reply by MikeFultonDev** [reply]
last page of 'man make' (note the <A9> for the copyright symbol - still have a few bugs to sort out)
![image](https://user-images.githubusercontent.com/61100689/220200034-a33f39d5-1a71-4b7f-a2ca-3eb853369037.png)

**Comment by lbdyck** [comment]
upgraded to git 2.40.1 and tried this - please advise.

![image](https://user-images.githubusercontent.com/42328411/235962397-d66915fc-9469-4d54-b5ed-cb9f268d8323.png)

**Reply by v1gnesh** [reply]
whence? lol that's new to me.
Only knew about `which`

**Comment by AnthonyGiorgio** [comment]
You probably need the zopen `man` port, as the z/OS built-in one can't handle formatted man pages.

**Reply by lbdyck** [reply]
![image](https://user-images.githubusercontent.com/42328411/235964291-98ab2353-b733-45b1-a32b-89902ad0f994.png)

**Reply by AnthonyGiorgio** [reply]
`zopen install man-db`

**Reply by lbdyck** [reply]
ty - i'll try that shortly.  appreciated.

**Comment by lbdyck** [comment]
Installed and then had to manually do the .env for it 

Now i get this 
![image](https://user-images.githubusercontent.com/42328411/235978707-f7ded4bc-ef13-4714-9fda-a8fa62dae01c.png)

**Reply by IgorTodorovskiIBM** [reply]
Thanks Lionel, a few problems here. It seems to be picking up the CI installed path of Jenkins rather than the one in the actual installed location.

If you change your pager to cat: `export PAGER=cat` does it get rendered correctly for you?

**Reply by lbdyck** [reply]
sorry - no

**Reply by lbdyck** [reply]
i'm on a tn3270 session under omvs - not using ssh

**Reply by v1gnesh** [reply]
I wonder, are you able to use bash within tn3270 omvs? 
Does that generally even work?

**Reply by lbdyck** [reply]
i can use bash and have the same challenge with man :(

**Comment by lbdyck** [comment]
Tried changing `_TAG_REDIR_OUT=txt` to just `_TAG_REDIR_OUT=` with no change.  changing `pager` did nothing other than there was no paging going on.

opened issue on man-db for `man man .`

**Comment by MikeFultonDev** [comment]
NEW: Turkish 'vim' (full UTF-8 support)

![image](https://github.com/ZOSOpenTools/meta/assets/61100689/31b1c7b1-2191-4486-aa6b-995d9969f7f5)

**Comment by MikeFultonDev** [comment]
i plan to add a section to the docs for this (unless someone else wants to?)

**Comment by MikeFultonDev** [comment]
https://github.com/ZOSOpenTools/meta/issues/443 opened
